src/functions.php

```diff
@@ -81,6 +81,35 @@ function captureCheckIn(string $slug, CheckInStatus $status, $duration = null, ?
     return SentrySdk::getCurrentHub()->captureCheckIn($slug, $status, $duration, $monitorConfig, $checkInId);
 }
 
+/**
+ * Execute the given callable while wrapping it in a monitor check-in.
+ *
+ * @param string             $slug          Identifier of the Monitor
+ * @param callable           $callback      The callable that is going to be monitored
+ * @param MonitorConfig|null $monitorConfig Configuration of the Monitor
+ *
+ * @return mixed
+ */
+function withMonitor(string $slug, callable $callback, ?MonitorConfig $monitorConfig = null)
+{
+    $checkInId = SentrySdk::getCurrentHub()->captureCheckIn($slug, CheckInStatus::inProgress(), null, $monitorConfig);
+
+    $status = CheckInStatus::ok();
+    $duration = 0;
+
+    try {
+        $start = microtime(true);
+        $result = $callback();
+        $duration = microtime(true) - $start;
+
+        return $result;
+    } catch (\Throwable $e) {
+        $status = CheckInStatus::error();
+    } finally {
```

Comment:
I believe that with a catch you do need to rethrow the error otherwise it's swallowed.

```suggestion
    } catch (\Throwable $e) {
        $status = CheckInStatus::error();
        
        throw $e;
    } finally {
```